b2share/modules/schemas/cli.py

```diff
@@ -46,3 +64,225 @@ def init(verbose):
+    community_schema = CommunitySchema.get_community_schema(comm.id, version)
+    if not community_schema:
+        raise click.ClickException("Community %s does not have a community schema" % comm.name)
+    community_schema_dict = json.loads(community_schema.community_schema)
+    props = community_schema_dict['properties']
+    click.secho("The following block schema versions are listed for community %s, community schema version %s" % (comm.name, "latest %d" % community_schema.version if not version else version))
+    for key in props:
+        click.secho("Block schema: %s, version url: %s" % (key, props[key]['$ref']))
+    
+@schemas.command()
+@with_appcontext
+@click.option('-v','--verbose', is_flag=True, default=False)
+@click.argument('community')
+@click.argument('block_schema_id')
+@click.option('--version')    
+def community_schema_add_block_schema_version(verbose, community, block_schema_id, version=None):
+    """Creates a new version of the community schema with this block_schema added. If called for the first time, creates the community schema and thereby activates the community."""
+    #fetch community
+    comm = get_community_by_name_or_id(community)
+    if not comm:
+        raise click.BadParameter("There is no community by this name or ID: %s" % community)
+    #fetch block schema
+    try:
+        val = UUID(block_schema_id, version=4)
+    except ValueError:
+        raise click.BadParameter("BLOCK_SCHEMA_ID is not a valid UUID (hexadecimal numbers and dashes e.g. fa52bec3-a847-4602-8af5-b8d41a5215bc )")
+    try:
+        block_schema = BlockSchema.get_block_schema(schema_id=block_schema_id)
+    except BlockSchemaDoesNotExistError:
+        raise click.BadParameter("No block_schema with id %s" % (block_schema_id))
+    if not version:
+        version = len(block_schema.versions)-1
+    if not version < len(block_schema.versions):
+        raise click.BadParameter("BlockSchema %s has %d versions, you specified version %d that does not exist" % (block_schema_id, len(block_schema.versions), version))
+    if verbose:
+        click.secho("Generating version URL for %s version %d" % (block_schema_id, version))
+    version_url = resolve_schemas_ref("$BLOCK_SCHEMA_VERSION_URL[%s::%d]" % (block_schema_id, version))
+    #fetch community_schema
+    community_schema = CommunitySchema.get_community_schema(comm.id)
+    #No community schema? Create one.
+    if not community_schema:
+        version_dict = {}
+        ref_dict = {}
+        ref_dict['$ref'] = version_url
+        version_dict[block_schema_id] = ref_dict
+        props_dict[block_schema_id] = version_dict
+        community_schema_dict = {}
+        community_schema_dict['properties'] = props_dict
+        community_schema_dict['$schema'] = "http://json-schema.org/draft-04/schema#"
+        community_schema_dict['type'] = "object"
+        community_schema_dict['additionalProperties'] = "false"
```

Comment:
Why not rewrite the entire block as a single dict, something like:
````
        community_schema_dict = {
               "$schema": "http://json-schema.org/draft-04/schema#",
                "type": "object",
                "properties": {
                    block_schema_id: {
                        "$ref": version_url,
                    }
                },
                "additionalProperties": false
````
I think it would be much more clear this way.